crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java

```diff
@@ -693,45 +713,154 @@ private String writeJSForLoop(StringBuilder bld, Path p, String varPrefix) {
+        "  var allEq=true;"+
+        "  for(var i=0i<this.%s.length;i++) { "+
+        "     if(this.%s[i] != this.%s[i]) { allEq=false; break; } "+
+        "  } "+
+        "  if(allEq) return true;"+
+        "}";
+
+    private static final String ARR_ARR_NEQ=
+        "if(this.%s.length==this.%s.length) { "+
+        "  var allEq=true;"+
+        "  for(var i=0i<this.%s.length;i++) { "+
+        "     if(this.%s[i] != this.%s[i]) { allEq=false; break; } "+
+        "  } "+
+        "  if(!allEq) return true;"+
+        "} else { return true; }";
+
+    private static final String ARR_ARR_LT=
+        "var n=this.%s.length<this.%s.length?this.%s.length:this.%s.length;"+
+        "var allOk=true;"+
+        "for(var i=0;i<n;i++) {"+
+        "  if(this.%s[i] >= this.%s[i]) { allOk=false;break;}"+
+        "}"+
+        "if(allOk&&(this.%s.length <= this.%s.length) return true;";
+
+    private static final String ARR_ARR_GT=
+        "var n=this.%s.length<this.%s.length?this.%s.length:this.%s.length;"+
+        "var allOk=true;"+
+        "for(var i=0;i<n;i++) {"+
+        "  if(this.%s[i] <= this.%s[i]) { allOk=false;break;}"+
+        "}"+
+        "if(allOk&&(this.%s.length >= this.%s.length) return true;";
+
+    private static final String ARR_ARR_LTE=
+        "var n=this.%s.length<this.%s.length?this.%s.length:this.%s.length;"+
+        "var allOk=true;"+
+        "for(var i=0;i<n;i++) {"+
+        "  if(this.%s[i] > this.%s[i]) { allOk=false;break;}"+
+        "}"+
+        "if(allOk&&(this.%s.length <= this.%s.length) return true;";
+
+    private static final String ARR_ARR_GTE=
+        "var n=this.%s.length<this.%s.length?this.%s.length:this.%s.length;"+
+        "var allOk=true;"+
+        "for(var i=0;i<n;i++) {"+
+        "  if(this.%s[i] < this.%s[i]) { allOk=false;break;}"+
+        "}"+
+        "if(allOk&&(this.%s.length >= this.%s.length) return true;";
+
+
+    private String writeArrayArrayComparisonJS(String field1,String field2,BinaryComparisonOperator op) {
+        switch(op) {
+        case _eq:
+            return String.format(ARR_ARR_EQ,field1,field2,field1,field1,field2);
+        case _neq:
+            return String.format(ARR_ARR_NEQ,field1,field2,field1,field1,field2);
+        case _lt:
+            return String.format(ARR_ARR_LT,field1,field2,field1,field2,field1,field2,field1,field2);
+        case _gt:
+            return String.format(ARR_ARR_GT,field1,field2,field1,field2,field1,field2,field1,field2);
+        case _lte:
+            return String.format(ARR_ARR_LTE,field1,field2,field1,field2,field1,field2,field1,field2);
+        case _gte:
+            return String.format(ARR_ARR_GTE,field1,field2,field1,field2,field1,field2,field1,field2);
+        }
+        return null;
+    }
+
+    private String writeArrayFieldComparisonJS(String field,String array,String op) {
+        return String.format("for(var i=0;i<this.%s.length;i++) { if(this.%s %s this.%s[i]) return true;}",array,field,op,array);
```

Comment:
I think this needs some testing.  Is the intent that if any element in the array satisfies "field op array[i]" that it's a match?